Description: Fix BTRFS+LUKS installations with unencrypted /boot
Author: TJ <hacker@iam.tj>
Origin: upstream, https://github.com/calamares/calamares/commit/60989290e02f20b56f7e90056a0239f000198cbb
Bug: https://github.com/calamares/calamares/issues/2281
Last-Update: 2024-02-12
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
diff --git a/src/modules/fstab/main.py b/src/modules/fstab/main.py
index 97e7e7486a..78cae63494 100755
--- a/src/modules/fstab/main.py
+++ b/src/modules/fstab/main.py
@@ -14,6 +14,7 @@
 
 import os
 import re
+import copy
 
 import libcalamares
 
@@ -201,7 +202,7 @@ def generate_fstab(self):
                     # so all subvolumes here should be safe to add to fstab
                     btrfs_subvolumes = libcalamares.globalstorage.value("btrfsSubvolumes")
                     for s in btrfs_subvolumes:
-                        mount_entry = partition
+                        mount_entry = copy.deepcopy(partition)
                         mount_entry["mountPoint"] = s["mountPoint"]
                         mount_entry["subvol"] = s["subvolume"]
                         dct = self.generate_fstab_line_info(mount_entry)
